daos-cortx/src/daos_kv_benchmark/kv_benchmarking.cc

```diff
@@ -0,0 +1,400 @@
+   char *key_buf = ( char * )calloc( key_size, sizeof( char ) ); // key buffer allocated
+
+   /* actual computation starts here */
+   for ( auto _ : state )
+   {
+      /* call daos_kv_put for state.range(2) times */
+      for ( int i = 0; i < num_keys; i++ )
+      {
+         state.PauseTiming( );
+
+         /* generate different key */
+         gen_key_names( key_buf, i, key_size );
+
+         state.ResumeTiming( );
+
+         /* actual function to mearsure time */
+         daos_kv_remove( oh, DAOS_TX_NONE, 0, key_buf, NULL );
+      }
+   }
+
+   /* free resources */
+   free( key_buf );
+
+   /* tear down */
+   tear_down( );
+}
+
+// bnechmark function is getting called here.
+static void kv_list_function( benchmark::State &state ) {
+   /* perform setup */
+   setup_main( );
+
+   unsigned int key_size = state.range( 0 );
+   unsigned int val_size = state.range( 1 );
+   unsigned int num_keys = state.range( 2 );
+   unsigned int nr_query = NR_QUERY;
+
+   /* allocate key and value buffers */
+   char *key_buf = ( char * )calloc( key_size, sizeof( char ) ); // key buffer allocated
+   char *val_buf = ( char * )calloc( val_size, sizeof( char ) ); // value buffer allocated
+   memset( val_buf, 'z', val_size - 1 );                         // populate with some random value.
+
+   char *rbuf = ( char * )calloc( val_size, sizeof( char ) ); // rbuf to check value
+
+   /* call daos_kv_put for state.range(2) times */
+   for ( int i = 0; i < num_keys; i++ )
+   {
+      /* generate different key */
+      gen_key_names( key_buf, i, key_size );
+
+      /* put keys and values */
+      daos_kv_put( oh, DAOS_TX_NONE, 0, ( char * )key_buf, val_size, val_buf, NULL );
+   }
+
+   /* actual computation starts here */
+   for ( auto _ : state )
+   {
+      char            *buf;
+      daos_key_desc_t kds[ NR_QUERY ];
+      daos_anchor_t   anchor = {
+         0
+      };
+      int             key_nr = 0;
+      d_sg_list_t     sgl;
+      d_iov_t         sg_iov;
+
+      buf = ( char * )calloc( KEY_DESC_BUF, sizeof( char ) );
+      d_iov_set( &sg_iov, buf, KEY_DESC_BUF );
+      sgl.sg_nr     = 1;
+      sgl.sg_nr_out = 0;
+      sgl.sg_iovs   = &sg_iov;
+
+      while ( !daos_anchor_is_eof( &anchor ) )
+      {
+         uint32_t nr = NR_QUERY;
+         int      rc;
+
+         memset( buf, 0, KEY_DESC_BUF );
+
+         rc = daos_kv_list( oh, DAOS_TX_NONE, &nr, kds, &sgl, &anchor, \
+                            NULL );
+         ASSERT( rc == 0, "KV list failed with %d", rc );
+
+         /* verify if returned number of descriptors are zero */
+         if ( nr == 0 )
+         {
+            continue;
+         }
+         else // if returned descriptors are non zero then compute each key and query value.
+         {
+            unsigned int offset = 0;
+
+            /* compute each key and fetch value */
+            for ( int i = 0; i < nr; i++ )
+            {
+               memset( key_buf, 0, key_size );
```

Comment:
exclude line 245 - 256 from timings to more accurately capture daos time

Reply:
done